apps/language_server/lib/language_server/providers/folding_range/line.ex

```diff
@@ -0,0 +1,47 @@
+defmodule ElixirLS.LanguageServer.Providers.FoldingRange.Line do
+  @moduledoc """
+  FoldingRange helpers for lines.
+  """
+
+  @type cell :: {non_neg_integer(), non_neg_integer() | nil}
+  @type t :: {String.t(), cell(), String.t()}
+
+  @spec format_string(String.t()) :: [cell()]
+  def format_string(text) do
+    text
+    |> text_to_lines()
+    |> embellish_lines_with_metadata()
+  end
+
+  @spec embellish_lines_with_metadata(String.t()) :: [String.t()]
+  defp text_to_lines(text) do
+    text
+    |> String.trim()
+    |> String.split("\n")
+  end
+
+  # If we think of the code text as a grid, this function finds the cells whose
+  # columns are the start of each row (line).
+  # Empty rows are represented as {row, nil}.
+  # We also grab the first character for convenience elsewhere.
+  @spec embellish_lines_with_metadata([String.t()]) :: [t()]
+  defp embellish_lines_with_metadata(lines) do
+    lines
+    |> Enum.with_index()
+    |> Enum.map(fn {line, row} ->
+      full_length = line |> String.length()
```

Comment:
Are those lengthts going to be returned to the client? If so, then you need to convert it to UTF16 char count - see ElixirLS.SourceFile for examples

Reply:
Oh nice! I'm actually seeing a few functions in `ElixirLS.LanguageServer.SourceFile` that I could be using.

Reply:
@lukaszsamson Yes, those lengths are being returned to the client. I think I've fixed it. LMK if not.

91252f0

Reply:
Looks good